src/specklepy/transports/server/server.py

```diff
@@ -73,9 +77,14 @@ def __init__(
                 warn(
                     SpeckleWarning(
                         "Unauthenticated Speckle Client provided to Server Transport"
-                        f" for {self.url}. Receiving from private streams will fail."
+                        f" for {url}. Receiving from private streams will fail."
                     )
                 )
+                self.account = get_default_account()
```

Comment:
what is the reason for providing a fallback to the default account?

Reply:
In case the ServerTransport is initiated via non-authenticated client (e.g. from Stream Wrapper of a public stream), the the self.account will stay None, and self.session.headers (line 101) will fail ("None has no attribute token")

Reply:
That is a StreamWrapper implementation issue, that we should fix there. Also the type of account is an Optional[Account], so we should do a None check on line 101 and only add an authorization header if we have an actual account.